### PR TITLE
Add CI/CD workflows

### PR DESCRIPTION
**Pull request by kaituo**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/anomaly-detection/issues/104

*Description of changes:*

This PR adds CI/CD workflows. CI workflow builds a zip file, installs that on the official ODFE docker image, and runs rest test case. CI workflow is triggered by branch pushing. CD workflow builds zip, rpm, deb files, and uploads those files to s3. CD workflow is triggered by tag creation.

This PR also increases model initialization waiting time as needed on DetectionResultEvalutationIT as it can be long on a remote cluster.

Testing done:
* ran the scripts on CI/CD workflow manually and tested it works.
* made sure local gradle build time does not increase due to my change in DetectionResultEvalutationIT.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java [hunk 1]
```diff
@@ -172,7 +172,22 @@ private void startDetector(
...
+        while (true) {
+            try {
+                getDetectionResult(detectorId, begin, end, client);
+                break;
+            } catch (Exception e) {
+                long duration = System.currentTimeMillis() - startTime;
+                // we wait at most 60 secs
+                if (duration < 60_000) {
+                    Thread.sleep(2_000);
+                } else {
```
Comment (wnbts) [inline src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java:186]: if it is timed out, is it better to fail fast?

File: src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java [hunk 2]
```diff
@@ -172,7 +172,22 @@ private void startDetector(
             } catch (Exception e) {}
         }
         Thread.sleep(5_000);
```
Comment (wnbts) [inline src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java:174]: this wait can be merged to the wait below now

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        run: |
+          cd ..
+          docker run -p 9200:9200 -d -p 9600:9600 -e "discovery.type=single-node" odfe-ad:test
+          sleep 90
+          curl -XGET http://localhost:9200/_cat/plugins
+
+      - name: Run AD Test
+        run: |
+          ./gradlew ':integTestRunner' --tests "com.amazon.opendistroforelasticsearch.ad.rest.*IT" -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster"
+          ./gradlew :integTestRunner --tests "com.amazon.opendistroforelasticsearch.ad.e2e.*IT"  -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster"
```
Comment (weicongs-amazon) [inline .github/workflows/CI.yml:63]: This won't run if the first one fails. just wanna confirm if this is what we expect

File: src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java (hunk 1 above)
Comment (kaituo) [reply src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java:186]: yeah, changed to throw exception.

File: src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java (hunk 2 above)
Comment (kaituo) [reply src/test/java/com/amazon/opendistroforelasticsearch/ad/e2e/DetectionResultEvalutationIT.java:174]: merged

File: .github/workflows/CI.yml (hunk 3 above)
Comment (kaituo) [reply .github/workflows/CI.yml]: good point.  Merged these 2 lines into one.

**Pull request merged**